lib/aws.go

```diff
@@ -0,0 +1,353 @@
+//
+// The last six parameters are as described for sign_aws_from_env.
+//
+// This corresponds to using [v4_creds.Signer.SignHTTP] using the result of a call
+// to [credentials.NewSharedCredentials].
+//
+// # Sign AWS from static credentials
+//
+// Returns a signed request based on credential details provided statically.
+//
+//	<map<string,dyn>>.sign_aws_from_static(<string>, <string>, <string>, <map<string,dyn>>) -> <map<string,dyn>>
+//	<map<string,dyn>>.sign_aws_from_static(<string>, <string>, <string>, <string>, <map<string,dyn>>) -> <map<string,dyn>>
+//
+// Examples:
+//
+//	req.sign_aws_from_static("id", "secret", "token", {"service": "service", … })   // return {"Body": "…
+//	req.sign_aws_from_static("id", "secret", "token", "source", {"service": "service", … })   // return {"Body": "…
+//
+// The last six parameters are as described for sign_aws_from_env.
+//
+// This corresponds to using [v4_creds.Signer.SignHTTP] using the result of a call
+// to [credentials.NewStaticCredentials] (nine parameter case) or
+// [credentials.NewStaticCredentialsFromCreds] (ten parameter case).
+func AWS() cel.EnvOption {
+	return cel.Lib(awsLib{})
+}
+
+type awsLib struct{}
+
+func (awsLib) CompileOptions() []cel.EnvOption {
+	return []cel.EnvOption{
+		cel.Function("sign_aws_from_env",
+			cel.MemberOverload(
+				"map_sign_aws_from_env_map",
+				[]*cel.Type{
+					mapKV,
+					mapStringDyn,
+				},
+				mapKV,
+				cel.FunctionBinding(catch(signFromEnv)),
+			),
+		),
+		cel.Function("sign_aws_from_shared",
+			cel.MemberOverload(
+				"map_sign_aws_from_shared_map",
+				[]*cel.Type{
+					mapKV,
+					cel.StringType, cel.StringType,
+					mapStringDyn,
+				},
+				mapKV,
+				cel.FunctionBinding(catch(signFromShared)),
+			),
+		),
+		cel.Function("sign_aws_from_static",
+			cel.MemberOverload(
+				"map_sign_aws_from_static_3_map",
+				[]*cel.Type{
+					mapKV,
+					cel.StringType, cel.StringType, cel.StringType,
+					mapStringDyn,
+				},
+				mapKV,
+				cel.FunctionBinding(catch(signFromCred)),
+			),
+			cel.MemberOverload(
+				"map_sign_aws_from_static_4_map",
+				[]*cel.Type{
+					mapKV,
+					cel.StringType, cel.StringType, cel.StringType, cel.StringType,
+					mapStringDyn,
+				},
+				mapKV,
+				cel.FunctionBinding(catch(signFromCred)),
+			),
+		),
+	}
+}
+
+func (awsLib) ProgramOptions() []cel.ProgramOption { return nil }
+
+func signFromEnv(args ...ref.Val) ref.Val {
+	const name = "sign_aws_from_env"
+	if len(args) != 2 {
+		return types.NewErr("no such overload for " + name)
+	}
+	request, ok := args[0].(traits.Mapper)
+	if !ok {
+		return types.ValOrErr(request, "no such overload for "+name)
+	}
+	creds, err := getEnv()
+	if err != nil {
+		return types.NewErr(err.Error())
+	}
+	cargs, ok := args[1].(traits.Mapper)
+	if !ok {
+		return types.NewErr("no such overload for sign_aws_from_creds")
+	}
+	return sign(request, creds, name, cargs)
```

Comment:
With `import "github.com/aws/aws-sdk-go-v2/config"` we can use the v2 SDK like this.

Apart from using the v2, the benefit is that is expands the scope of the env var support to cover more auth uses cases like using `AWS_ROLE_ARN ` with `AWS_WEB_IDENTITY_TOKEN_FILE` (which is common in EKS).

```suggestion
	creds, err := config.NewEnvConfig()
	if err != nil {
		return types.NewErr(err.Error())
	}
	cargs, ok := args[1].(traits.Mapper)
	if !ok {
		return types.NewErr("no such overload for sign_aws_from_creds")
	}
	return sign(request, creds.Credentials, name, cargs)
```